packages/react-hooks/src/useOnEvent.ts

```diff
@@ -0,0 +1,28 @@
+import { on } from '@uifabric/utilities';
+import * as React from 'react';
+
+/**
+ * Hook to attach an event handler on mount and handle cleanup
+ * @param element- Element (or ref to an element) to attach the event handler to
+ * @param eventName- The event to attach a handler for
+ * @param callback- The handler for the event
+ * @param useCapture- Whether or not to attach the handler for the capture phase
+ */
+export function useOnEvent<TElement extends Element>(
+  element: React.RefObject<TElement | undefined | null> | TElement | Window | undefined | null,
+  eventName: string,
+  callback: (ev: Event) => void,
+  useCapture?: boolean,
+) {
+  React.useEffect(() => {
+    if (element && 'current' in element) {
+      element = element.current;
+    }
+    if (!element) {
+      return;
+    }
+
+    const dispose = on(element, eventName, callback, useCapture);
+    return dispose;
+  }, [element, eventName, callback, useCapture]);
```

Comment:
If someone passes in a callback that mutates every render, this is going to result in a lot of extra attaching/detaching events. Not sure if that's a problem practically, and I don't think we should be artificially preventing it inside the hook (in case the user does want to change the callback), but maybe at least add a note in the doc comment?

Reply:
I changed the code to stash the callback in a ref, to obviate the need to cleanup/reattach in such cases.